src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java

```diff
@@ -128,7 +128,9 @@ public boolean move(String path, String target) {
         if (!file.exists() && !dest.exists()) {
             return true; // moving an undefined resource
         }
-
+        if( !file.exists() ){
+            throw new IllegalStateException("Unable to find " + path + " to move to " + target);
+        }
```

Comment:
This seems wrong/redundant. The check `!file.exists() && !dest.exists()` will often return true before reaching here. We will only get the IllegalStateException if `!file.exists() && dest.exists()`, and the cach block after Files.move already throws this exception anyway.